src/geometry/regionbasedadjacency.h

```diff
@@ -34,27 +34,35 @@ class RegionBasedAdjacency
 {
 public:
     RegionBasedAdjacency();
+    RegionBasedAdjacency(const RegionBasedAdjacency &other);
+    RegionBasedAdjacency &operator=(const RegionBasedAdjacency &other);
+    virtual ~RegionBasedAdjacency() {}
+
+#ifdef LIBGEODECOMP_WITH_CPP14
+    RegionBasedAdjacency(RegionBasedAdjacency &&other) = default;
+    RegionBasedAdjacency &operator=(RegionBasedAdjacency &&other) = default;
+#endif // LIBGEODECOMP_WITH_CPP14
 
     /**
      * Insert a single edge (from, to) to the graph
      */
-    void insert(int from, int to);
+    virtual void insert(int from, int to);
```

Comment:
To enable users to derive from this class and define their own adjacencies. We could also add an own Adjacency interface, that would probably be cleaner.